README.md

```diff
@@ -115,123 +113,117 @@ GoogleSignin.configure({
+import { GoogleSignin, statusCodes } from 'react-native-google-signin';
 
-```js
-const user = GoogleSignin.currentUser();
-// user is null if not signed in
-```
-
-#### - signIn
-
-Prompt the modal to let the user signin into your application
-
-```js
+// Somewhere in your code
 signIn = async () => {
   try {
-    const user = await GoogleSignin.signIn();
-    this.setState({ user });
+    const userInfo = await GoogleSignin.signIn();
+    this.setState({ userInfo });
   } catch (error) {
-    if (error.code === 'CANCELED') {
+    if (error.code === statusCodes.SIGN_IN_CANCELLED) {
       // user cancelled the login flow
+    } else if (error.code === statusCodes.IN_PROGRESS) {
+      // operation (f.e. sign in) is in progress already
     } else {
       // some other error happened
     }
   }
 };
 ```
 
-#### - getAccessToken (Android Only)
+#### - `signInSilently()`
 
-Obtain the user access token.
+May be called eg. in the `componentDidMount` of your main component. This method returns the [current user](#3-userinfo) if they already signed in and `null` otherwise.
+
+To see how to handle errors read [`signIn()` method](#--signin)
 
 ```js
-GoogleSignin.getAccessToken()
-  .then(token => {
-    console.log(token);
-  })
-  .catch(err => {
-    console.log(err);
-  });
+getCurrentUser = async () => {
+  try {
+    const userInfo = await GoogleSignin.signInSilently();
+    this.setState({ userInfo });
+  } catch (error) {
+    console.error(error);
+  }
+};
 ```
 
-#### - signOut
+#### - `signOut()`
 
-remove user session from the device
+Remove user session from the device.
 
 ```js
-_signOut = async () => {
+signOut = async () => {
   try {
     await GoogleSignin.revokeAccess();
     await GoogleSignin.signOut();
-    this.setState({ user: null });
+    this.setState({ user: null }); // Remember to remove the user from your app's state as well
   } catch (error) {
-    this.setState({
-      error,
-    });
+    console.error(error)
   }
 };
 ```
 
-#### - revokeAccess
+#### - `revokeAccess()`
 
-remove your application from the user authorized applications
+Remove your application from the user authorized applications.
 
 ```js
 GoogleSignin.revokeAccess()
   .then(() => {
     console.log('deleted');
   })
-  .catch(err => {
-    console.log(err);
+  .catch(error => {
+    console.error(error);
   });
 ```
 
-### 3. User
+#### - `statusCodes`
+
+These are useful when determining which kind of error has occured during sign in process. Import `statusCodes` along with `GoogleSignIn`. Under the hood these constants are derived from native GoogleSignIn error codes and are platform specific and can be changed. Always prefer to compare `error.code` to `statusCodes.SIGN_IN_CANCELLED` or `statusCodes.IN_PROGRESS` and not relying on raw value of the `error.code`.
+
+Name | Description 
+--- | ---
+`SIGN_IN_CANCELLED` | When user cancels the sign in flow
+`IN_PROGRESS` | Trying to invoke another sign in flow when previous one has not yet finished
```

Comment:
> Trying to invoke another sign in flow when previous one has not yet finished

the error can be thrown not only with `signIn` but with several other calls as well, and those calls are even different per-platform 😆 so it would be good to make this a little more generic